**1.Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.**

In [1]:
from math import factorial

Число сочетаний

$$C_{n}^k =\frac{n!}{k!(n - k)!}$$

In [2]:
def combinations(n, k):
    # Число сочетаний из  𝑛  элементов по  𝑘  элементов в каждом (в сочетаниях порядок не важен):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Число размещений

$$C_{n}^k =\frac{n!}{(n - k)!}$$

In [3]:
def arrangements(n, k):
    # число размещений из  𝑛  элементов по  𝑘  элементов в каждом. При размещениях порядок важен
    return int(factorial(n) / factorial(n - k))

Число перестановок

$$P_{n} = n!$$

In [4]:
def permutations(n):
    # Число перестановок из  𝑛  элементов
    return int(factorial(n))

a) Найти вероятность того, что все карты – крести
Количество карт масти Крести n= 52/4=13
Количество комбинаций из 4 карт масти крести
$$C_{13}^4 =\frac{n!}{k!(n - k)!}= \frac{13!}{4!(13-4)!}=715, $$
$$n=13, k=4$$
Возможно 715 комбинаций с 4 картами масти крести. Количество карт масти крести = 14.

In [5]:
print(int(52/4))
C_13_4=combinations(13, 4)
C_13_4

13


715

Количество комбинаций из 4 карт из колоды в 52 карты$$C_{52}^4 =\frac{n!}{k!(n - k)!}= \frac{52!}{4!(52-4)!}=270725, $$$$n=52, k=4$$
Возможно 270725 комбинаций с 4 картами из колоды 52 карты.

In [6]:
C_52_4=combinations(52, 4)
C_52_4

270725

Вероятнисть того, что все 4 карты крести$$P = \frac{C_{13}^4}{C_{52}^4}= \frac{715}{270725}=0.00264 $$

In [7]:
P=C_13_4/C_52_4
P

0.0026410564225690276

Ответ: Вероятнисть того, что все 4 карты крести P=0,00264

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз
Событие B={появится хотя бы один туз}
52-4=48 - количество карт без 4 тузов
Вероятность события B - P(B):
$$P(B) = 1-\frac{C_{48}^4}{C_{52}^4}= 1 - \frac{194580}{270725}=1 - 0.719 = 0.281$$
$C_{48}^4 = 194580$ - Количество комбинаций по 4 карты из колоды в 52 карт из которой убрали 4 туза
$C_{52}^4 = 270725$ - Количество комбинаций по 4 карты из колоды в 52 карт
1 - полная вероятность
${C_{48}^4} / {C_{52}^4}= 0.719$ - Вероятность того, что среди карт не окажется ни одного туза

In [8]:
C_48_4=combinations(48, 4)
C_48_4

194580

In [9]:
C_52_4=combinations(52, 4)
C_52_4

270725

In [10]:
P_b = 1 - C_48_4/C_52_4
P_b

0.2812632745405854

Ответ: Вероятность того, что все 4 карты крести P=0,281

**2.На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?**

In [12]:
# Функция для вычисления количества сочетаний
def сombination (n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [13]:
n = сombination(10, 3)
print(f'Количество сочетаний из 3-х цифр от 0 до 9: {n}')

Количество сочетаний из 3-х цифр от 0 до 9: 120


In [14]:
p = 1/n
print(f'Вероятность угадать код с первой попытки: {p}')

Вероятность угадать код с первой попытки: 0.008333333333333333


Ответ: Вероятность угадать код с первой попытки: 0.00833 или 0.833%

**3.В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**

Решение:
Тянем 1-ю деталь, с вероятностью 9/15=3/5 вытянем окрашенную деталь.
Тянем 2-ю деталь. Вероятность вытянуть 2-ю окрашенную деталь = 8/14 = 4/7.
Тянем 3-ю деталь. Вероятность вытянуть 3-ю окрашенную деталь = 7/13.
Вероятность 3 раза подряд вытянуть окрашенную деталь = (3/5)(4/7)(7/13) = 0.18461538461538457 или около 18,5%

In [15]:
(3/5)*(4/7)*(7/13)

0.18461538461538457

Ответ: Вероятность 3 раза подряд вытянуть окрашенные детали = 18.5%

**4.В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**

In [16]:
n = сombination(100, 2)
print(f'Общее число исходов, из 2-х билетов: {n}')

Общее число исходов, из 2-х билетов: 4950


In [17]:
m = сombination(2, 2)
print(f'Количество положительных исходов: {m}')

Количество положительных исходов: 1


In [18]:
p = m/n
print(f'Вероятность вытащить два выигрышных билета: {p}')

Вероятность вытащить два выигрышных билета: 0.00020202020202020202


Ответ: Вероятность угадать код с первой попытки: 0.00020 или 0.020%